src/main/javascript/service/cat-api-service.js

```diff
@@ -323,6 +323,20 @@ function CatApiServiceProvider() {
             function $getCatApiService($http, catConversionService) {
             var catApiService = {};
 
+            /**
+             * This method allows to define (dynamic) endpoints after the configuration phase.
+             * @param {object} [settings] a new {EndpointConfig} will be created with the given settings
+             * @returns {CatApiEndpoint}
+             */
+            catApiService.$endpoint = function (settings) {
+                var name = 'dynamic-endpoint';
+                if (!_.isUndefined(settings)) {
+                    return new CatApiEndpoint(_urlPrefix, new EndpointConfig(name, settings), $http, catConversionService);
```

Comment:
i'm not yet quite convinced if setting a hardcoded endpoint name and just returning the instance is the best solution

how about refactoring the method as follows

```javascript
var dynamicEndpoints = {};

catApiService.dynamicEndpoint = function (name, settings) {
    if (_.isUndefined(dynamicEndpoints[name])) {
        dynamicEndpoints[name] = // new endoint
    }

    return dynmaicEndpoints[name];
}
```